### PR TITLE
Add validation for reserved filename pattern misuse

### LINKED ISSUE #1459: Validate is not failing products with reserved names
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When validating the attached collection, the `collection_overview.xml` label passes validation even though the name `collection_*.xml` is reserved. This is not a collection label, since the actual collection label is `collection_data.xml`

### 🕵️ Expected behavior

I expected the `collection_overview.xml` label to fail with an error indicating that the name is reserved.

### 📜 To Reproduce

Using the attached data:

Run the validator:
```bash
validate --skip-content-validation -t reserved_not_failing
```

Observe that `collection_overview.xml` passes:

```
  PASS: file:/Users/jessestone/Desktop/neese-validate-20251216/reserved_not_failing/data/collection_overview.xml (urn:nasa:pds:orex.derived_radio_science:data:collection_overview::1.0)
      WARNING  [warning.label.context_ref_mismatch]   line 35: Context reference name mismatch. Value: 'Origins, Spectral Interpretation, Resource Identification, Security, Regolith Explorer (OSIRIS-REx)' Expected one of: '[orex, The Origins, Spectral Interpretation, Resource Identification, Security, Regolith Explorer (OSIRIS-REx) Mission, OSIRIS-REx, Origins, Spectral Interpreation, Resource Identification, Security, Regolith Explorer (OSIRIS-REx) Mission]'
      WARNING  [warning.label.context_ref_mismatch]   line 45: Context reference name mismatch. Value: 'OSIRIS-REx Spacecraft' Expected one of: '[orex, Origins, Spectral Interpreation, Resource Identification, Security, Regolith Explorer (OSIRIS-REx) Spacecraft, The Origins, Spectral Interpretation, Resource Identification, Security, Regolith Explorer (OSIRIS-REx) Spacecraft, OSIRIS-REx]'
      WARNING  [warning.label.context_ref_mismatch]   line 61: Context reference name mismatch. Value: 'Touch-and-Go Camera Suite (TAGCAMS)' Expected one of: '[TAGCAMS, The Touch-and-Go Camera Suite (TAGCAMS) aboard the OSIRIS-REx spacecraft]'
      WARNING  [warning.label.context_ref_mismatch]   line 69: Context reference name mismatch. Value: 'OSIRIS_REx Camera Suite (OCAMS)' Expected one of: '[The OSIRIS-REx Camera Suite (OCAMS) aboard the OSIRIS-REx spacecraft, OCAMS, The OSIRIS-REx Camera Suite Suite (OCAMS) aboard the OSIRIS-REx spacecraft]'
        3 product validation(s) completed
```

### 🖥 Environment Info

Ubuntu Linux 22.04
Java 17

### 📚 Version of Software Used

validate-4.0.8

### 🩺 Test Data / Additional context

[reserved_not_failing.zip](https://github.com/user-attachments/files/24201116/reserved_not_failing.zip)

[reserved_not_failing.txt](https://github.com/user-attachments/files/24201118/reserved_not_failing.txt)

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### 🎉 Integration & Test

_No response_

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Per PDS4 Standards Reference 6C.1.3, certain file name patterns (collection_*, bundle_*) are reserved for specific product types. This PR adds validation to detect when a file uses a reserved pattern but the product type doesn't match the expected type.

**What was the issue?**
Files like `collection_overview.xml` (which is a Product_Document) were passing validation even though they use the reserved `collection_*` filename pattern, which should only be used for Product_Collection products.

**What changed?**
- Added new `RESERVED_FILE_NAME_MISMATCH` warning type for reserved filename pattern violations
- Implemented `checkReservedFilename()` method in `LabelValidationRule` that:
  - Detects filenames matching reserved patterns (collection_*, bundle_*)
  - Examines the actual product type in the XML
  - Reports a WARNING if there's a mismatch
- Added Cucumber test case with real-world test data from the issue

**Why WARNING instead of ERROR?**
Based on discussion in the issue, the consensus was to use WARNING level (not ERROR) to flag potential issues while allowing teams flexibility, as there are existing missions using similar naming patterns for documentation files.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 95%

## ⚙️ Test Data and/or Report

**Automated Test Added:**
- Cucumber test scenario added to `src/test/resources/features/3.7.x.feature`
- Test data copied from issue #1459: `src/test/resources/github1459/data/`
- Test validates that `collection_overview.xml` triggers the new warning

**Test Execution:**
```bash
mvn test -Dtest=CucumberTest -Dcucumber.filter.tags='@3.7.x' -Dcucumber.filter.name='.*1459.*'
```

**Test Output:**
```
Scenario Outline: NASA-PDS/validate#1459-1
  ✔ Given validate issue 1459, test 1, and test data at "github1459/data"
  ✔ When execute validate with "--skip-content-validation -t {datasrc}/collection_overview.xml"
  ✔ Then compare to the expected outcome

1 scenarios (1 passed)
3 steps (3 passed)
```

**Manual Validation:**
```bash
./validate --skip-content-validation -t collection_overview.xml

# Output includes:
WARNING [warning.file.reserved_name_mismatch] File name 'collection_overview.xml' 
uses the reserved 'collection_' pattern but the product type is not Product_Collection.
Reserved file names should only be used for their intended product types (PDS4 SR 6C.1.3).
```

## ♻️ Related Issues

Fixes #1459

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: src/main/java/gov/nasa/pds/tools/validate/rule/pds4/LabelValidationRule.java [hunk 1]
```diff
@@ -175,6 +175,49 @@ private void flagBadFilename(URL target) {
     }
   }
 
+  /**
+   * Check if a file uses a reserved filename pattern (collection_*, bundle_*, etc.) but the product
+   * type doesn't match the expected type. According to PDS4 Standards Reference 6C.1.3, certain file
+   * names are reserved for specific purposes.
+   */
+  private void checkReservedFilename(URL target) {
+    String filename = FilenameUtils.getName(target.toString());
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/tools/validate/rule/pds4/LabelValidationRule.java]: `FilenameUtils.getName(target.toString())` relies on string conversion of the URL; using `target.toURI()` + `File` may be safer and more semantically correct

File: src/main/java/gov/nasa/pds/tools/validate/rule/pds4/LabelValidationRule.java [hunk 2]
```diff
@@ -175,6 +175,49 @@ private void flagBadFilename(URL target) {
...
+    if (!isReservedCollectionPattern && !isReservedBundlePattern) {
+      // Filename doesn't match any reserved patterns, no validation needed
+      return;
+    }
+
+    // Determine the actual product type by examining the XML content
+    gov.nasa.pds.tools.validate.TargetExaminer examiner =
+        new gov.nasa.pds.tools.validate.TargetExaminer();
+    boolean isCollectionProduct = examiner.isTargetCollectionType(target, true);
+    boolean isBundleProduct = examiner.isTargetBundleType(target, true);
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/tools/validate/rule/pds4/LabelValidationRule.java]: Possible performance issue? Does `TargetExaminer` appears to re-parse the label? Since `validateLabel()` already processes the label, maybe we can reuse the existing product class+type instead of re-deriving it here?

**Review by nutjob4life — APPROVED** [review]
A couple of inline comments but also: approved!

Details:
```
[INFO] Tests run: 32, Failures: 0, Errors: 0, Skipped: 23
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12:38 min
```

**Comment by jordanpadams** [comment]
Thanks for the review! I've addressed both comments:

**Comment 1 (filename extraction):** Changed from `FilenameUtils.getName(target.toString())` to `new File(target.toURI()).getName()` for safer and more semantically correct filename extraction.

**Comment 2 (performance/re-parsing):** This is a great catch! I've moved the validation from `LabelValidationRule` to `RegisterTargets` instead. This is much better architecturally because:

- **No duplicate parsing**: `RegisterTargets.registerTargets()` already determines product type via `Utility.getTargetType()` which uses `TargetExaminer` with caching. We piggyback on that existing work instead of parsing twice.
- **Single source of truth**: Product type is determined once, and our validation uses the same already-computed `TargetType`.
- **Follows existing patterns**: `RegisterTargets` is where targets are classified and registered, making it the natural place to validate filename/type consistency.

The implementation now checks reserved filenames right after determining product type, with zero additional I/O overhead. Much cleaner!

Also fixed the import - no more fully qualified class name `gov.nasa.pds.tools.validate.TargetExaminer`. 😄
